lib/hsa/mcwamp_hsa.cpp

```diff
@@ -494,15 +499,18 @@ class HSABarrier : public Kalmar::KalmarAsyncOp {
     bool isDispatched;
     hsa_wait_state_t waitMode;
 
+
     std::shared_future<void>* future;
 
     // prior dependencies
     // maximum up to 5 prior dependencies could be associated with one
     // HSABarrier instance
     int depCount;
 
-
 public:
+    uint16_t  header;
+    hc::memory_scope _acquire_scope;
```

Comment:
@bensander we could use some documentation here for `_acquire_scope`, but later on in `enqueuemarker` we would take another parameter `releaseScope`. The difference between them deserve to be logged.